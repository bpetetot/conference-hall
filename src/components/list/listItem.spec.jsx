/* eslint-env jest */
import React from 'react'
import { shallow } from 'enzyme'
import { ListItem } from './index'

describe('components/listItem', () => {
  it('should render', () => {
    expect(
      shallow(
        <ListItem
          title="title"
          subtitle="subtitle"
          info="info"
        />,
      ),
    ).toMatchSnapshot()
  })
  it('should render clickable if onSelect is provided', () => {
    const clickMock = jest.fn()
    const wrapper = shallow(
      <ListItem
        title="title"
        subtitle="subtitle"
        info="info"
        onSelect={clickMock}
      />,
    )
    expect(wrapper.find('.clickable').length).toEqual(1)
    wrapper.find('.clickable').simulate('click')
    expect(clickMock).toHaveBeenCalled()
    expect(wrapper).toMatchSnapshot()
  })
})
