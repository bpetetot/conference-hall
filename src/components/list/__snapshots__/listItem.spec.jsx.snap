// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/listItem should render 1`] = `
<div
  className="list-item"
  role="button"
>
  <div>
    <div
      className="list-item-title"
    >
      title
    </div>
    <div
      className=" list-item-subtitle"
    >
      subtitle
    </div>
  </div>
  <div
    className="list-item-info"
  >
    info
  </div>
</div>
`;

exports[`components/listItem should render clickable if onSelect is provided 1`] = `
<div
  className="list-item clickable"
  onClick={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  role="button"
>
  <div>
    <div
      className="list-item-title"
    >
      title
    </div>
    <div
      className=" list-item-subtitle"
    >
      subtitle
    </div>
  </div>
  <div
    className="list-item-info"
  >
    info
  </div>
</div>
`;
